From 77a8421c90d17fcd71884d94cb85678b51f3b7b1 Mon Sep 17 00:00:00 2001
From: John Zhuge <jzhuge@cloudera.com>
Date: Thu, 11 Aug 2016 13:36:08 -0700
Subject: [PATCH 1717/2748] HDFS-8521. Add VisibleForTesting annotation to
 BlockPoolSlice#selectReplicaToDelete. (cmccabe)

(cherry picked from commit 806e407ac8896c4e669dba1fcf86fa5d6fee7c6d)
(cherry picked from commit 12497b0543176306edbd5cd8660a59b1e2e3a1c8)

Conflicts:
	hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/BlockPoolSlice.java

(cherry picked from commit a7058caa729bbaf5ae0e46d2372c40d06b936337)

Change-Id: I0d901d0720571a0daa3bf1ef64ff6a42d5fa658d
---
 .../datanode/fsdataset/impl/BlockPoolSlice.java    |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/BlockPoolSlice.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/BlockPoolSlice.java
index 46f51eb..9022ae3 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/BlockPoolSlice.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/BlockPoolSlice.java
@@ -52,6 +52,8 @@
 import org.apache.hadoop.util.ShutdownHookManager;
 import org.apache.hadoop.util.Time;
 
+import com.google.common.annotations.VisibleForTesting;
+
 /**
  * A block pool slice represents a portion of a block pool stored on a volume.  
  * Taken together, all BlockPoolSlices sharing a block pool ID across a 
@@ -547,6 +549,7 @@ ReplicaInfo resolveDuplicateReplicas(
     return replicaToKeep;
   }
 
+  @VisibleForTesting
   static ReplicaInfo selectReplicaToDelete(final ReplicaInfo replica1,
       final ReplicaInfo replica2) {
     ReplicaInfo replicaToKeep;
-- 
1.7.9.5

